test/helper/sending.bash

```diff
@@ -45,57 +47,127 @@ function _send_email() {
       ( '--server' ) SERVER=${2:?--server given but no argument} ; shift 2 ;;
       ( '--port' )   PORT=${2:?--port given but no argument}     ; shift 2 ;;
       ( '--data' )
-        local TEMPLATE_FILE="/tmp/docker-mailserver-test/emails/${2:?--data given but no argument provided}.txt"
-        FINAL_SWAKS_OPTIONS+=('--data')
-        FINAL_SWAKS_OPTIONS+=('-')
-        FINAL_SWAKS_OPTIONS+=('<')
-        FINAL_SWAKS_OPTIONS+=("${TEMPLATE_FILE}")
+        ADDITIONAL_SWAKS_OPTIONS+=('--data')
+        local FILE_PATH="/tmp/docker-mailserver-test/emails/${2:?--data given but no argument provided}"
+        if _exec_in_container_bash "[[ -e ${FILE_PATH} ]]"; then
+          ADDITIONAL_SWAKS_OPTIONS+=("@${FILE_PATH}")
+        else
+          ADDITIONAL_SWAKS_OPTIONS+=("'${2}'")
+        fi
         shift 2
         ;;
-      ( * ) ADDITIONAL_SWAKS_OPTIONS+=("${1}") ; shift 1 ;;
+      ( * ) ADDITIONAL_SWAKS_OPTIONS+=("'${1}'") ; shift 1 ;;
     esac
   done
 
-  _run_in_container_bash "swaks --server ${SERVER} --port ${PORT} --ehlo ${EHLO} --from ${FROM} --to ${TO} ${ADDITIONAL_SWAKS_OPTIONS[*]} ${FINAL_SWAKS_OPTIONS[*]}"
+  _run_in_container_bash "swaks --server '${SERVER}' --port '${PORT}' --ehlo '${EHLO}' --from '${FROM}' --to '${TO}' ${ADDITIONAL_SWAKS_OPTIONS[*]}"
+}
+
+# Sends a mail from localhost (127.0.0.1) to a container. To send
+# a custom email, create a file at `test/files/<TEST FILE>`,
+# and provide `<TEST FILE>` as an argument to this function.
+#
+# Parameters include all options that one can supply to `swaks`
+# itself. The `--data` parameter expects a relative path from `emails/`
+# where the contents will be implicitly provided to `swaks` via STDIN.
+#
+# This functions performs an implicit `assert_success` to check whether
+# the e-mail transaction was successful. If this is not desirable, use
+# `_send_email_unchecked`.
+#
+# ## Attention
+#
+# This function assumes `CONTAINER_NAME` to be properly set (to the container
+# name the command should be executed in)!
+#
+# This function will just send the email in an "asynchronous" fashion, i.e. it will
+# send the email but it will not make sure the mail queue is empty after the mail
+# has been sent.
```

Comment:
```suggestion
# This function will send the email in an "asynchronous" fashion,
# it will return without waiting for the Postfix mail queue to be emptied.
```